implementations/rust/ockam/ockam_api/src/influxdb/lease_token.rs

```diff
@@ -33,6 +35,34 @@ pub struct LeaseToken {
     pub status: TokenStatus,
 }
 
+impl Encodable for LeaseToken {
```

Comment:
Have you tried to implement Encodable for every Encode?
The cons would be that you need to create a new type if the root message is encoded both in cbor and bare

```
impl<T: Encode<()> + CborLen<()>> Encodable for T {
    fn encode(self) -> Result<Encoded> {
        cbor_encode_preallocate(self)
    }
}
````

Reply:
I did it on purpose in order to not tie the `Encodable` trait to any implementation. It used to be tied to `Serializable` before that PR.

We currently have a mix of `cbor` and `serde` at the moment and this way it becomes a conscious choice on how you want to serialize your message data. Also note that we can't specify both a generic encoding for `T: Encode` and `T: Serializable` in Rust.

Reply:
Perhaps a macro derive to implement Encodable with cbor would help in this case? It would be explicit as to how the trait is implemented and it would keep the trait isolated.